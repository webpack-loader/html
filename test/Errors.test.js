const loader = require('../src/index.js')
const webpack = require('@webpack-utilties/test')

describe('Errors', () => {
  test('Loader Error', async () => {
    const config = {
      loader: {
        test: /\.html$/,
        options: {
          minimize: true      
        }
      }
    }

    const stats = await webpack('error.js', config)
    const { source } = stats.toJson().modules[1]

    // eslint-disable-next-line
    const err = () => eval(source)

    expect(err).toThrow()
    expect(err).toThrowErrorMatchingSnapshot()
  })

  test('Validation Error', async () => {
    const config = {
      loader: {
        test: /\.html$/,
        options: {
          template: 1
        }
      }
    }

    const stats = await webpack('error.js', config)
    const { errors } = stats.toJson()

    errors.forEach((error) => expect(error).toMatchSnapshot())
  })
})
