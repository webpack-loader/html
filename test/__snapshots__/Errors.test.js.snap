// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Errors Loader Error 1`] = `
"Module build failed: HTML Loader: 

tree.forEach is not a function
"
`;

exports[`Errors Validation Error 1`] = `
"./fixture.html
Module build failed: ValidationError: HTML Loader Invalid Options

options.template should be boolean,string

 @ ./error.js 1:0-34 3:15-19"
`;
